src/Servers/HttpSys/src/RequestProcessing/RequestContext.FeatureCollection.cs

```diff
@@ -598,6 +601,34 @@ IHeaderDictionary IHttpResponseTrailersFeature.Trailers
 
     CancellationToken IConnectionLifetimeNotificationFeature.ConnectionClosedRequested { get; set; }
 
+    bool IHttpSysRequestTimingFeature.TryGetTimestamp(HttpSysRequestTimingType timestampType, out long timestamp)
+    {
+        int index = (int)timestampType;
+        if (index < Request.RequestTimestamps.Count && Request.RequestTimestamps[index] > 0)
+        {
+            timestamp = Request.RequestTimestamps[index];
+            return true;
+        }
+
+        timestamp = default;
+        return false;
+    }
+
+    bool IHttpSysRequestTimingFeature.TryGetElapsedTime(HttpSysRequestTimingType startingTimestampType, HttpSysRequestTimingType endingTimestampType, out TimeSpan elapsed)
+    {
+        var timingFeature = (IHttpSysRequestTimingFeature)this;
+        if (timingFeature.TryGetTimestamp(startingTimestampType, out long startTimestamp) && timingFeature.TryGetTimestamp(endingTimestampType, out long endTimestamp))
+        {
+            // TODO: Switch to using TimeProvider which is in preview 4
+            // elapsed = TimeProvider.System.GetElapsedTime(startTimestamp, endTimestamp);
+            elapsed = new TimeSpan((long)((endTimestamp - startTimestamp) * ((double)TimeSpan.TicksPerSecond / Stopwatch.Frequency)));
```

Comment:
Ha, not sure how I missed the fact that Stopwatch had that method also :) I was going to use TimeProvider.System so it would have been fine but using Stopwatch makes more sense.